elliptic-curve/src/point/non_identity.rs

```diff
@@ -103,6 +108,66 @@ impl<P> AsRef<P> for NonIdentity<P> {
     }
 }
 
+impl<const N: usize, P> BatchNormalize<[Self; N]> for NonIdentity<P>
+where
+    P: Curve + BatchNormalize<[P; N], Output = [P::AffineRepr; N]>,
+{
+    type Output = [NonIdentity<P::AffineRepr>; N];
+
+    fn batch_normalize(points: &[Self; N]) -> [NonIdentity<P::AffineRepr>; N] {
+        debug_assert_eq!(size_of::<P>(), size_of::<NonIdentity<P>>());
+        debug_assert_eq!(align_of::<P>(), align_of::<NonIdentity<P>>());
+        debug_assert_eq!(
+            size_of::<P::AffineRepr>(),
+            size_of::<NonIdentity<P::AffineRepr>>()
+        );
+        debug_assert_eq!(
+            align_of::<P::AffineRepr>(),
+            align_of::<NonIdentity<P::AffineRepr>>()
+        );
+
+        #[expect(unsafe_code, reason = "`NonIdentity` is `repr(transparent)`")]
+        let points: &[P; N] = unsafe { mem::transmute(points) };
```

Comment:
I think a pointer cast provides more compile-time checks, possibly including some of your `debug_assert`s above:

```suggestion
        let points = unsafe { &*(points as *const [Self; N] as *const [P; N]) };
```

Unfortunately I don't think you can use the [`cast`](https://doc.rust-lang.org/std/primitive.pointer.html#method.cast) function here.